source/index.ts

```diff
@@ -0,0 +1,746 @@
+	*/
+	onDidAnyChange(callback: () => unknown): () => unknown {
+		if (typeof callback !== 'function') {
+			throw new TypeError(`Expected \`callback\` to be of type \`function\`, got ${typeof callback}`);
+		}
+
+		const getter: () => unknown = () => this.store;
+
+		return this.handleChange(getter, callback);
+	}
+
+	handleChange(getter: () => unknown, callback: (newValue: unknown, oldValue: unknown) => void): () => void {
+		let currentValue = getter();
+
+		const onChange: () => unknown = () => {
+			const oldValue = currentValue;
+			const newValue = getter();
+
+			try {
+				// TODO: Use `util.isdeepEqual` when targeting Node.js 10
+				assert.deepEqual(newValue, oldValue);
+			} catch (_) {
+				currentValue = newValue;
+				callback.call(this, newValue, oldValue);
+			}
+		};
+
+		this.events?.on('change', onChange);
+		return () => this.events?.removeListener('change', onChange);
+	}
+
+	encryptData(data: string | Buffer): string | Buffer {
+		if (!this.encryptionKey) {
+			return data;
+		}
+
+		try {
+			// Check if an initialization vector has been used to encrypt the data
+			if (data.slice(16, 17).toString() === ':') {
+				const initializationVector = data.slice(0, 16);
+				const password = crypto.pbkdf2Sync(this.encryptionKey, initializationVector.toString(), 10000, 32, 'sha512');
+				const decipher = crypto.createDecipheriv(encryptionAlgorithm, password, initializationVector);
+				const slicedData: any = data.slice(17);
+				return Buffer.concat([decipher.update(slicedData), decipher.final()]);
+			}
+
+			// Legacy decryption without initialization vector
+			const decipher = crypto.createDecipher(encryptionAlgorithm, this.encryptionKey);
+			const legacyData: any = data;
+			return Buffer.concat([decipher.update(legacyData), decipher.final()]);
+		} catch (_) {
+			return data;
+		}
+	}
+
+	get size(): number {
+		return Object.keys(this.store).length;
+	}
+
+	get store(): any {
+		try {
+			const data: string | Buffer = fs.readFileSync(this.path, this.encryptionKey ? null : 'utf8');
+			const dataString: string | Buffer = this.encryptData(data);
+			const deserializedData = this.deserialize && this.deserialize(dataString);
+			this._validate(deserializedData);
+			return Object.assign(plainObject(), deserializedData);
+		} catch (error) {
+			if (error.code === 'ENOENT') {
+				this._ensureDirectory();
+				return plainObject();
+			}
+
+			if (this._options.clearInvalidConfig && error.name === 'SyntaxError') {
+				return plainObject();
+			}
+
+			throw error;
+		}
+	}
+
+	set store(value: any) {
+		this._ensureDirectory();
+
+		this._validate(value);
+		this._write(value);
+
+		this.events?.emit('change');
+	}
+
+	* [Symbol.iterator](): IterableIterator<[string, any]> {
+		for (const [key, value] of Object.entries(this.store)) {
+			yield [key, value];
+		}
+	}
+}
+
+export type Serializer = (...args: unknown[]) => string;
+export type Deserializer = (arg: string | Buffer) => unknown;
```

Comment:
```suggestion
export type Deserializer = (data: string) => unknown;
```

It should not accept `Buffer`.